cpp/tests/rolling/collect_ops_test.cpp

```diff
@@ -1230,3 +1230,605 @@ TYPED_TEST(TypedCollectListTest, GroupedTimeRangeRollingWindowOnStructsWithMinPe
 
   CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected_result->view(), result_with_nulls_excluded->view());
 }
+
+struct CollectSetTest : public cudf::test::BaseFixture {
```

Comment:
The tests look fairly comprehensive. Good job here.

We should add tests here for the following:
1. In the floats/doubles case, we need to test the behaviour when float list rows contain multiple NaN elements (with different NaN values.
2. Explicit tests for when the input row contains non-comparable types. E.g. `struct`, `list`. We should error out in those cases.

Reply:
Please also add tests for the case when the input columns are sliced column. This is important.

Reply:
Hi @mythrocks @ttnghia, I added tests for NaN, non-comparable types and sliced columns.